src/Sylius/Bundle/CoreBundle/Resources/config/app/messenger.yaml

```diff
@@ -1,14 +1,30 @@
 framework:
     messenger:
-        # Uncomment this (and the failed transport below) to send failed messages to this transport for later handling.
-
         transports:
-        # https://symfony.com/doc/current/messenger.html#transport-configuration
-            main: '%env(MESSENGER_TRANSPORT_DSN)%'
+            main: 
+                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
+            catalog_promotion_removal: 
+                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
+                failure_transport: 'catalog_promotion_removal_failed'
+                options:
+                    queue_name: 'catalog_promotion_removal'
+            catalog_promotion_removal_failed:
+                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
+                options:
+                    queue_name: 'catalog_promotion_removal_failed'
+            failed: 
+                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
+                options:
+                    queue_name: 'failed'
+                    
+        failure_transport: failed
+        reset_on_message: true
 
         routing:
             'Sylius\Bundle\CoreBundle\CatalogPromotion\Command\UpdateCatalogPromotionState': main
             'Sylius\Bundle\CoreBundle\CatalogPromotion\Command\ApplyCatalogPromotionsOnVariants': main
+            'Sylius\Bundle\CoreBundle\CatalogPromotion\Command\TerminateCatalogPromotion': main
+            'Sylius\Bundle\CoreBundle\CatalogPromotion\Command\RemoveCatalogPromotion': catalog_promotion_removal
             'Sylius\Component\Promotion\Event\CatalogPromotionCreated': main
             'Sylius\Component\Promotion\Event\CatalogPromotionEnded': main
             'Sylius\Component\Promotion\Event\CatalogPromotionUpdated': main
```

Comment:
I know we already have transport called `main` but there is a risk someone can have the same transport name in their app. IMO we should prefix transport names with `sylius_` – similar to what we have in DB table names.

```suggestion
            main: 
                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
            sylius_catalog_promotion_removal: 
                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
                failure_transport: 'sylius_catalog_promotion_removal_failed'
                options:
                    queue_name: 'sylius_catalog_promotion_removal'
            sylius_catalog_promotion_removal_failed:
                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
                options:
                    queue_name: 'sylius_catalog_promotion_removal_failed'
                    
        reset_on_message: true

        routing:
            'Sylius\Bundle\CoreBundle\CatalogPromotion\Command\UpdateCatalogPromotionState': main
            'Sylius\Bundle\CoreBundle\CatalogPromotion\Command\ApplyCatalogPromotionsOnVariants': main
            'Sylius\Bundle\CoreBundle\CatalogPromotion\Command\TerminateCatalogPromotion': main
            'Sylius\Bundle\CoreBundle\CatalogPromotion\Command\RemoveCatalogPromotion': sylius_catalog_promotion_removal
            'Sylius\Component\Promotion\Event\CatalogPromotionCreated': main
            'Sylius\Component\Promotion\Event\CatalogPromotionEnded': main
            'Sylius\Component\Promotion\Event\CatalogPromotionUpdated': main
```

I would also be for deprecating `main` transport and using `sylius_main` instead in a separate PR. WDYT?
And is `failed` transport needed as we have `sylius_catalog_promotion_removal_failed`?

Reply:
It makes sense to me, any other opinions? :) @GSadee @Zales0123 @NoResponseMate?

Reply:
On the other hand, if processed messages have the same priority for both Sylius and end app custom tasks, is there a  need to introduce another transport?

Reply:
@coldic3 , "And is failed transport needed as we have sylius_catalog_promotion_removal_failed"
Failed transport was introduced as a fallback for the `main` transport, Agree that this is not the case of removing catalog promotions but does give the opportunity of manually resolving failed ( by default 3x at row ) `main` transport messages by the administrator instead of abandoning the that unsuccessful jobs

Reply:
> On the other hand, if processed messages have the same priority for both Sylius and end app custom tasks, is there a need to introduce another transport?

IMO In theory it's not needed to introduce another transport in such a case but we don't know the transport name in the end app. It might be `main` but it doesn't have to be.


> @coldic3 , "And is failed transport needed as we have sylius_catalog_promotion_removal_failed" Failed transport was introduced as a fallback for the `main` transport, Agree that this is not the case of removing catalog promotions but does give the opportunity of manually resolving failed ( by default 3x at row ) `main` transport messages by the administrator instead of abandoning the that unsuccessful jobs

I'm in favor of having `failed` transport for `main` but changing `failure_transport` globally (for all transports) we interfere not only in Sylius behavior itself but also in the end app behavior. Therefore maybe specifying this failed transport only for `main` transport will be better? 🤔

Reply:
I've changed the configuration according to your proposal ✋